util.py

```diff
@@ -109,6 +109,19 @@ def make_dataloader(input_ids, labels, attention_masks, batch_size=16, shuffle=T
                             )
     return dataloader
 
+def split_episode(batch, ratio=.5):
+    batch_size = batch[0].size()[0]
+    support_size = round(batch_size*ratio)
+
+    support_idx_pos = np.random.choice(np.arange(0, batch_size, 2), int(np.ceil(support_size/2)), replace=False)
+    support_idx_neg = np.random.choice(np.arange(1, batch_size, 2), int(np.floor(support_size/2)), replace=False)
```

Comment:
With the Episodic sampler this is the case. However in the case with three or more labels this approach would be flawed (not guaranteed to yield equal label distribution within support/query set although approximately true)
But as Phillip mentioned we should make the support and query set two completely separate data sets anyway (otherwise we will get previously seen examples in the query set)
If we make these changes then this function won't be needed anymore

Reply:
Cool, let's wait for those changes then.